src/univariate/discrete/poissonbinomial.jl

```diff
@@ -0,0 +1,113 @@
+
+# Computes the pdf of a poisson-binomial random variable using
+# fast fourier transform
+# 
+#     Hong, Y. (2013). 
+#     On computing the distribution function for the Poisson binomial 
+#     distribution. Computational Statistics and Data Analysis, 59, 41–51.
+#
+function poissonbinomial_pdf_fft(p::AbstractArray)
+    n = length(p)
+    ω = 2. * pi / (n + 1)
+
+    x = Array(Complex{Float64}, n+1)
+    lmax = int(ceil(n/2))
+    x[1] = 1./(n+1)
+    for l=1:lmax
+        logz = 0.
+        argz = 0.
+        for j=1:n
+            zjl = 1 - p[j] + p[j] * cos(ω*l) + im * p[j] * sin(ω * l)
```

Comment:
Rather than multiply `ω` by `pi`, you can use `cospi`/`sinpi` here: it will be more accurate for values like `cos(pi / 2)`.